src/libsyntax/ast.rs

```diff
@@ -1,33 +1,97 @@
-// The Rust abstract syntax tree.
+//! The Rust abstract syntax tree module.
+//!
+//! This module contains common structures forming the language AST.
+//! Two main entities in the module are [`Item`] (which represents an AST element with
+//! additional metadata), and [`ItemKind`] (which represents a concrete type and contains
+//! information specific to the type of the item).
+//!
+//! Other module items that worth mentioning:
+//! - [`Ty`] and [`TyKind`]: A parsed Rust type.
+//! - [`Expr`] and [`ExprKind`]: A parsed Rust expression.
+//! - [`Pat`] and [`PatKind`]: A parsed Rust pattern. Patterns are often dual to expressions.
+//! - [`Stmt`] and [`StmtKind`]: An executable action that does not return a value.
+//! - [`FnDecl`], [`FnHeader`] and [`Param`]: Metadata assotiated with function declaration.
+//! - [`Generics`], [`GenericParam`] and [`WhereClause`]: Metadata assotiated with generics.
+//! - [`EnumDef`] and [`Variant`]: Enum declaration.
+//! - [`Lit`] and [`LitKind`]: Literal expressions.
+//! - [`MacroDef`], [`MacStmtStyle`], [`Mac`], [`MacDelimeter`]: Macro definition and invocation.
+//! - [`Attribute`]: Metadata assotiated with item.
+//! - [`UnOp`], [`UnOpKind`], [`BinOp`], [`BinOpKind`]: Unary and binary operators.
+//!
+//! [`Item`]: struct.Item.html
+//! [`ItemKind`]: enum.ItemKind.html
+//! [`Ty`]: struct.Ty.html
+//! [`TyKind`]: enum.TyKind.html
+//! [`Pat`]: struct.Pat.html
+//! [`PatKind`]: enum.PatKind.html
+//! [`Stmt`]: struct.Stmt.html
+//! [`StmtKind`]: enum.StmtKind.html
+//! [`FnDecl`]: struct.FnDecl.html
+//! [`FnHeader`]: struct.FnHeader.html
+//! [`Param`]: struct.Param.html
+//! [`Generics`]: struct.Generics.html
+//! [`GenericParam`]: struct.GenericParam.html
+//! [`WhereClause`]: struct.WhereClause.html
+//! [`EnumDef`]: struct.EnumDef.html
+//! [`Variant`]: struct.Variant.html
+//! [`Lit`]: struct.Lit.html
+//! [`LitKind`]: enum.LitKind.html
+//! [`MacroDef`]: struct.MacroDef.html
+//! [`MacStmtStyle`]: enum.MacStmtStyle.html
+//! [`Mac`]: struct.Mac.html
+//! [`MacDelimeter`]: enum.MacDelimeter.html
+//! [`Attribute`]: struct.Attribute.html
+//! [`UnOp`]: type.UnOp.html
+//! [`UnOpKind`]: enum.UnOpKind.html
+//! [`BinOp`]: type.BinOp.html
+//! [`BinOpKind`]: enum.BinOpKind.html
 
 pub use GenericArgs::*;
 pub use UnsafeSource::*;
 pub use crate::util::parser::ExprPrecedence;
 
-use crate::parse::token::{self, DelimToken};
-use crate::ptr::P;
-use crate::source_map::{dummy_spanned, respan, Spanned};
-use crate::tokenstream::TokenStream;
-
-use rustc_target::spec::abi::Abi;
 pub use rustc_target::abi::FloatTy;
-
-use syntax_pos::{Span, DUMMY_SP, ExpnId};
-use syntax_pos::symbol::{kw, sym, Symbol};
 pub use syntax_pos::symbol::{Ident, Symbol as Name};
 
+use crate::{
+    parse::token::{self, DelimToken},
```

Comment:
Style: could you avoid multi-line imports and remove these formatting changes?
(Sorting imports is ok.)